### PR TITLE
python3Packages.lcov-cobertura: fix missing distutils at runtime

### LINKED ISSUE #349331: lcov-cobertura is missing a dependency on distutils
### Describe the bug
```ShellSession
$ nix run nixpkgs#python3Packages.lcov-cobertura
Traceback (most recent call last):
  File "/nix/store/i4a4asfarhffj9rsaghmiq8a81zll34n-python3.12-lcov-cobertura-2.0.2/bin/.lcov_cobertura-wrapped", line 6, in <module>
    from lcov_cobertura.lcov_cobertura import main
  File "/nix/store/i4a4asfarhffj9rsaghmiq8a81zll34n-python3.12-lcov-cobertura-2.0.2/lib/python3.12/site-packages/lcov_cobertura/lcov_cobertura.py", line 21, in <module>
    from distutils.spawn import find_executable
ModuleNotFoundError: No module named 'distutils'

```

### Steps To Reproduce
1. See above

### Expected behavior
It should run successfully.

### Notify maintainers
@mweinelt 
@wucke13 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.2, NixOS, 24.11 (Vicuna), 24.11.20241006.c31898a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.8`
 - channels(root): `"nixos"`
 - channels(jkolb): `""`
 - nixpkgs: `/nix/store/rs4fjbnw4qx7ns2hzzrz2iz52va7vs5z-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by kjeremy**
Fixes #349331

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kjeremy** [comment]
Now I get:

```console
[jkolb@nixos:~/nixpkgs]$ nix run .#legacyPackages.x86_64-linux.python3Packages.lcov-cobertura
warning: Git tree '/home/jkolb/nixpkgs' is dirty

    Converts LCOV coverage data to Cobertura-compatible XML for reporting.

    Usage:
        lcov_cobertura.py lcov-file.dat
        lcov_cobertura.py lcov-file.dat -b src/dir -e test.lib -o path/out.xml

    By default, XML output will be written to ./coverage.xml
```

File: pkgs/development/python-modules/lcov-cobertura/default.nix [hunk 1]
```diff
@@ -17,6 +18,9 @@ buildPythonPackage rec {
     hash = "sha256-xs40e/PuZ/jV0CDNZiYmo1lM8r5yfMY0qg0R+j9/E3Q=";
   };
 
+  build-system = [ setuptools ];
+  dependencies = [ setuptools ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/lcov-cobertura/default.nix:22]:
```suggestion

  dependencies = [ distutils ];
```

**Comment by kjeremy** [comment]
@mweinelt  for some reason your suggestion caused the file to be unparseable so I did it manually and force pushed. I don't actually need to depend on `distutils` though. Just having `depencencies = [ setuptools ];` seemed to work. I don't know python well enough to know the difference.

**Comment by mweinelt** [comment]
Because you didn't import distutils at the time.

File: pkgs/development/python-modules/lcov-cobertura/default.nix [hunk 2]
```diff
@@ -17,6 +19,9 @@ buildPythonPackage rec {
     hash = "sha256-xs40e/PuZ/jV0CDNZiYmo1lM8r5yfMY0qg0R+j9/E3Q=";
   };
 
+  build-system = [ setuptools ];
+  dependencies = [ distutils ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lcov-cobertura/default.nix:23]: we could also apply https://github.com/eriwen/lcov-to-cobertura-xml/pull/51

**Pull request merged**